helm/default-apps-aws/templates/apps.yaml

```diff
@@ -22,60 +22,42 @@ spec:
     secret:
       name: {{ $.Values.clusterName }}-kubeconfig
       namespace: {{ $.Release.Namespace}}
-  {{- with .clusterValues }}
-  {{- if or .configMap .secret }}
   config:
-  {{- if .configMap }}
     configMap:
       name: {{ $.Values.clusterName }}-cluster-values
       namespace: {{ $.Release.Namespace }}
-  {{- end }}
-  {{- if .secret }}
-    secret:
-      name: {{ $.Values.clusterName }}-cluster-values
-      namespace: {{ $.Release.Namespace }}
-  {{- end }}
-  {{- end }} 
-{{- end }}
-{{- with (get $.Values.userConfig $key) }}
+  {{- with (get $.Values.userConfig $key) }}
   userConfig:
-  {{- if .configMap }}
     configMap:
       name: {{ $.Values.clusterName }}-{{ $appName }}-user-values
       namespace: {{ $.Release.Namespace }}
   {{- end }}
-  {{- if .secret }}
-    secret:
-      name: {{ $.Values.clusterName }}-{{ $appName }}-user-values
-      namespace: {{ $.Release.Namespace }}
-  {{- end }}
-  {{- end }}
+{{- end }}
 
-{{- with (get $.Values.userConfig $key) }}
-{{- if .configMap }}
 ---
 apiVersion: v1
 kind: ConfigMap
 metadata:
   labels:
     {{- include "labels.common" $ | nindent 4 }}
-  name: {{ $.Values.clusterName }}-{{ $appName }}-user-values
+  name: {{ $.Values.clusterName }}-cilium-user-values
   namespace: {{ $.Release.Namespace }}
 data:
-  {{- (tpl (.configMap | toYaml | toString) $) | nindent 2 }}
-{{- end }}
-{{- if .secret }}
----
-apiVersion: v1
-kind: Secret
-metadata:
-  labels:
-    {{- include "labels.common" $ | nindent 4 }}
-  name: {{ $.Values.clusterName }}-{{ $appName }}-user-values
-  namespace: {{ $.Release.Namespace }}
-stringData:
-  {{- (tpl (.secret | toYaml | toString) $) | nindent 2 }}
-{{- end }}
-{{- end }}
-
-{{- end }}
+  values: |
+    ipam:
+      mode: kubernetes
+    k8sServiceHost: api.{{ $.Values.baseDomain }}
```

Comment:
Is this change entirely so you can use template strings here?

If so, can we not keep the dynamicness and pass the data to `tpl` ?

Reply:
See https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-tpl-function

Reply:
At least at the moment I did the change, there were no apps making use of the dynamic config/secrets other than the cilium configmap. It feels simpler to just write the manifests that we want, instead of trying to generate them. Happy to discuss it if you think otherwise.

Reply:
Dynamic means we can override values without the new of a new chart release. We can override app settings with a configmap uservalues.

Reply:
ok, if you think it's necessary I'll try to make it work with the tpl function.

Reply:
I think it's worth having if we can. But if too difficult then I'm not that fussed.

Reply:
It seems we were already using `tpl`, so it was already prepared for it https://github.com/giantswarm/default-apps-aws/pull/82